java/src/main/native/OrtJniUtil.c

```diff
@@ -600,427 +588,473 @@ size_t copyPrimitiveArrayToJava(JNIEnv *jniEnv, ONNXTensorElementDataType onnxTy
-    // Extract tensor type
-    OrtTensorTypeAndShapeInfo* tensorInfo;
-    checkOrtStatus(jniEnv,api,api->GetTensorTypeAndShape(tensor,&tensorInfo));
-    ONNXTensorElementDataType value;
-    checkOrtStatus(jniEnv,api,api->GetTensorElementType(tensorInfo,&value));
-
-    // Get the element count of this tensor
-    size_t length;
-    checkOrtStatus(jniEnv,api,api->GetTensorShapeElementCount(tensorInfo,&length));
+  jdoubleArray outputArray = NULL;
+  // Extract tensor type
+  OrtTensorTypeAndShapeInfo* tensorInfo = NULL;
+  OrtErrorCode code = checkOrtStatus(jniEnv,api,api->GetTensorTypeAndShape(tensor, &tensorInfo));
+  if (code == ORT_OK) {
+    ONNXTensorElementDataType value = ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED;
+    code = checkOrtStatus(jniEnv,api,api->GetTensorElementType(tensorInfo, &value));
+    if ((code == ORT_OK) && (value == ONNX_TENSOR_ELEMENT_DATA_TYPE_DOUBLE)) {
+        // Get the element count of this tensor
+        size_t length = 0;
+        code = checkOrtStatus(jniEnv,api,api->GetTensorShapeElementCount(tensorInfo, &length));
+        if (code == ORT_OK) {
+            // Extract the values
+            uint8_t* arr = NULL;
+            code = checkOrtStatus(jniEnv,api,api->GetTensorMutableData((OrtValue*)tensor, (void**)&arr));
+            if (code == ORT_OK) {
+                // Create the java array and copy to it.
+                outputArray = (*jniEnv)->NewDoubleArray(jniEnv, safecast_size_t_to_jsize(length));
+                int64_t consumed = copyPrimitiveArrayToJava(jniEnv, value, arr, outputArray);
+                if (consumed == -1) {
+                    outputArray = NULL;
+                }
+            }
+        }
+    }
     api->ReleaseTensorTypeAndShapeInfo(tensorInfo);
-
-    // Extract the values
-    uint8_t* arr;
-    checkOrtStatus(jniEnv,api,api->GetTensorMutableData((OrtValue*)tensor,(void**)&arr));
-
-    // Create the java array and copy to it.
-    jdoubleArray outputArray = (*jniEnv)->NewDoubleArray(jniEnv,safecast_size_t_to_jsize(length));
-    copyPrimitiveArrayToJava(jniEnv, value, arr, outputArray);
-    return outputArray;
+  }
+  return outputArray;
 }
 
 jobject createJavaTensorFromONNX(JNIEnv *jniEnv, const OrtApi * api, OrtAllocator* allocator, OrtValue* tensor) {
-    // Extract the type information
-    OrtTensorTypeAndShapeInfo* info;
-    checkOrtStatus(jniEnv,api,api->GetTensorTypeAndShape(tensor, &info));
-
-    // Construct the TensorInfo object
-    jobject tensorInfo = convertToTensorInfo(jniEnv, api, info);
+  // Extract the type information
+  OrtTensorTypeAndShapeInfo* info;
+  OrtErrorCode code = checkOrtStatus(jniEnv, api, api->GetTensorTypeAndShape(tensor, &info));
+  if (code != ORT_OK) {
+    return NULL;
+  }
 
-    // Release the info object
+  // Construct the TensorInfo object
+  jobject tensorInfo = convertToTensorInfo(jniEnv, api, info);
+  if (tensorInfo == NULL) {
     api->ReleaseTensorTypeAndShapeInfo(info);
```

Comment:
> api->ReleaseTensorTypeAndShapeInfo(info); [](http://example.com/codeflow?start=3&length=42)

This seems to be release in both cases error and success. Perhaps, we can release it before check for null.

Reply:
Fixed.